dask/dataframe/tests/test_dataframe.py

```diff
@@ -306,19 +306,94 @@ def test_describe(method, test_values):
     assert_eq(df.describe(), ddf.describe(split_every=2, percentiles_method=method))
 
 
-@pytest.mark.parametrize('method', ['tdigest', 'dask'])
-def test_describe_empty(method):
-    if method == 'tdigest':
-        pytest.importorskip('crick')
-    # https://github.com/dask/dask/issues/2326
-    ddf = dd.from_pandas(pd.DataFrame({"A": ['a', 'b']}), 2)
-    with pytest.raises(ValueError) as rec:
-        ddf.describe(percentiles_method=method)
-    assert 'DataFrame contains only non-numeric data.' in str(rec)
+@pytest.mark.parametrize('include,exclude,percentiles', [
+    (None, None, None),
+    ('all', None, None),
+    (['number'], None, [0.25, 0.5]),
+    ([np.timedelta64], None, None),
+    (['number', 'object'], None, [0.25, 0.75]),
+    (None, ['number', 'object'], None),
+    (['object', 'datetime', 'bool'], None, None)
+])
+def test_describe(include, exclude, percentiles):
+    data = {
+        'a': ["aaa", "bbb", "bbb", None, None, "zzz"] * 2,
+        'c': [None, 0, 1, 2, 3, 4] * 2,
+        'd': [None, 0, 1] * 4,
+        'e': [pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 07:56:23.858694'),
+              pd.Timestamp('2017-05-09 05:59:58.938999'),
+              None,
+              None] * 2,
+        'f': [np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D'),
+              None,
+              None,
+              np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D')] * 2,
+        'g': [True, False, True] * 4
+    }
+
+    # Arrange
+    df = pd.DataFrame(data)
+    ddf = dd.from_pandas(df, 2)
+
+    # Act
+    desc_ddf = ddf.describe(include=include, exclude=exclude, percentiles=percentiles)
+    desc_df = df.describe(include=include, exclude=exclude, percentiles=percentiles)
+    desc_ddf_computed = desc_ddf.compute()
+
+    # TODO: for timedelta columns there's overflow in var function, see #4233
+    # causing std to be nan, ignoring this cell
+    if 'f' in desc_ddf_computed:
+        desc_df['f']['std'] = None
+        desc_ddf_computed['f']['std'] = None
```

Comment:
I'd just drop this column, so you can avoid computing before passing to `assert_eq`. Something like:

```python
if 'f' in desc_df.columns:
    desc_df = desc_df.drop('f', axis=1)
    desc_ddf = desc_ddf.drop('f', axis=1)

assert_eq(desc_df, desc_ddf)
```

Reply:
@jcrist thank you for the comments! My concern is, if we drop whole column the output for timedelta will not be tested at all. What if I workaround it with this:
```python
 if 'f' in desc_ddf._meta:
        with pytest.raises(AssertionError) as info:
            assert_eq(desc_ddf, desc_df)
            msg = "DataFrame.iloc[:, 2] are different"
            assert msg in str(info.value)
```

It's very ugly for sure but at least the remaining input is validated. What do you think?